src/sentry/models/projectownership.py

```diff
@@ -100,11 +117,31 @@ def get_autoassign_owners(cls, project_id, data, limit=2):
 
         Returns a tuple of (auto_assignment_enabled, list_of_owners).
         """
+        from sentry.models import ProjectCodeOwners
+
         with metrics.timer("projectownership.get_autoassign_owners"):
             ownership = cls.get_ownership_cached(project_id)
-            if not ownership:
+            codeowners = ProjectCodeOwners.get_codeowners_cached(project_id)
+
+            if not (ownership or codeowners):
                 return False, []
 
+            if not ownership:
+                ownership = cls(project_id=project_id)
```

Comment:
@MeredithAnya Yes we create an ownership record if it doesn't exist. This is just incase of corrupt data. Test here: https://github.com/getsentry/sentry/blob/27baba1742357c440a86dff3f1c9e549d5cd937f/tests/sentry/models/test_projectownership.py#L78-L80

Reply:
@MeredithAnya hmm both patterns work but with the suggested pattern it will require more logic wrt default settings across the frontend and backend. The current pattern separates the settings stuff from codeowners. The potential downside of the current pattern is more "empty" records in the db which may affect the efficiency of the negative IssueOwners cache. I think for now, the current pattern is ok.

+1 on the logging and docstring updates

Reply:
@NisanthanNanthakumar I guess what Im trying to say is that your tests already prove that codeowners works without having issue owners. 

The auto_assignment default is set to False, which means that us creating a dummy account has the same effect as not creating it at all.  I can go into my settings a toggle on auto_assignment for an empty issue owners, because when I do that it creates an issue owners record if I haven't already. I don't think there are any UI changes that would have to be made